### PR TITLE
Private Files: Always send X-Private header

### PR DESCRIPTION
**Pull request by mjangda**
Even if the file is public, send a value of false to avoid traps when we hit an error

### Private Files: Always send X-Private header

To harden the feature and avoid unexpected issues when the ACL endpoint may hit an error (e.g. fatal), we now will always send the X-Private header when the endpoint successfully processes the file. This way, we can differentiate between successful and unsuccessful requests and handle them appropriately upstream.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- n/a This change has relevant documentation additions / updates (if applicable).
- [x] I've created a changelog description that aligns with the provided examples. 

## Steps to Test

Covered by unit tests.

### CODE REVIEW

File: files/acl/acl.php [hunk 1]
```diff
@@ -71,19 +71,22 @@ function send_visibility_headers( $file_visibility, $file_path ) {
...
 		case FILE_IS_PRIVATE_AND_ALLOWED:
 			$status_code = 202;
-			$header = 'X-Private: true';
+			$is_private = true;
 			break;
 
 		case FILE_IS_PRIVATE_AND_DENIED:
 			$status_code = 403;
+			$is_private = true;
 			$header = 'X-Private: true';
```
Comment (alexcriss) [inline files/acl/acl.php:90]: I think we can probably get rid of this too :)

**Review by alexcriss — CHANGES_REQUESTED** [review]
Thanks a lot for this, I left one minor silly comment!

File: files/acl/acl.php (hunk 1 above)
Comment (mjangda) [reply files/acl/acl.php]: Of course! Good catch!

**Review by nickdaugherty — APPROVED** [review]
LGTM

**Comment by rinatkhaziev** [comment]
r1380-stacks

**Pull request merged**